compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -660,6 +660,12 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
 
       origJsClass.memberDefs.foreach {
         case fdef: js.FieldDef =>
+          /* Private fields are both used when building the instance *and* kept
+           * as declarations in the abstract JS type, so that we can validate
+           * `JSPrivateSelect` nodes.
+           */
+          if (fdef.name.isInstanceOf[js.Ident])
+            staticMembers += fdef
```

Comment:
Rename `staticMembers` to `classDefMembers`? Or at the very least comment that the field is *not* static.